game-frontend/src/service/game/context.ts

```diff
@@ -11,14 +11,17 @@ import {
 import reducer from './reducer';
 
 const USER_ID_QUERY_PARAM = 'userId';
+export const ROLE_LOCAL_STORAGE_KEY = 'role';
 const useUserId = createPersistedState(USER_ID_QUERY_PARAM);
+const usePersistRole = createPersistedState(ROLE_LOCAL_STORAGE_KEY);
 
 export const WebsocketContext = createContext<ContextState>(
   emptyWebsocketContext
 );
 
 export function useWebsocket(): ContextState {
   const [userId, setUserId] = useUserId<UserId>(() => uuidv4());
+  const [role, setRole] = usePersistRole<Role | null>(() => null);
```

Comment:
The callback for `usePersistRole` is not required, simply do `usePersistRole<Role | null>(null);` in this case.